### ISSUE TITLE
OAuth tutorial review

### ISSUE THREAD

**Issue by gazzadc** [ibm-apiconnect/getting-started#96]
Tutorial reviewed: https://github.com/ibm-apiconnect/getting-started/blob/master/bluemix/2b/README.md

**Prerequisites** - Need to avoid positional words like "below" as they fail accessibility checks. Just say "you must have completed the following tutorial:"

There is a main heading **Create an OAuth Provider API and select your OAuth scheme** and yet the steps that follow go on to do much more than just create an OAuth Provider API and select an OAuth scheme, and there are no other main headings in the tutorial.

Step 1

- Need to say what UI we should be in, and how to launch it (looks like it's the API Manager UI?).
- Need to give explicit instructions instead of "From Drafts/APIs". For example, the user might be on the Dashboard page with the navigation pane collapsed, so you need to give the precise instructions for opening the Drafts > APIs page.
- The two occurrences of "select" should be "click". Select is specifically for when you choose an item from a set of alternatives, but for straightforward clicking of buttons and icons it's "click".
- "Select Create" -> "Select Create API" (do you know if the tutorial was tested, as this should have been picked up?)
- The panel is called "OAuth 2", with a space (wouldn't it be better just to click "OAuth 2" in the navigation pane rather than scroll down?)
- The screen capture needs to come **before** the Save step - screen captures need to show the user what they should be seeing in front of them at that precise point in the tutorial. (by the way, the screen capture doesn't match what I am seeing, I see the grants vertically aligned rather than horizontal)
- You have a set of bullet items under step 1, but they aren't in any way subitems to that first step, they are separate steps in their own right, therefore they need to continue the numerical sequencing. Alternatively (probably better), have a main Step 1 a called "Create an OAuth 2.0 provider API", then the steps that you have here need to be substeps under that, and lettered (a. b. c. ....). Note that you shouldn't use bullets for a sequence of steps, bullets are only for lists. Step sequences should be numbered, or lettered if they are substeps under a numbered step.

Step 2

- Here, the bulleted items **are** substeps, so make them lettered (a. b. c. ....)
- Need to be more explicit than "Switch over to your Weather Provider API" (at this point, the user is still on the design page for the OAuth provider API)
- Re "add a new definition for OAuth", need to be more explicit - the user has to click the icon to add a new security definition, then select the OAuth definition type.
- Re "Update the Flow to Application", it's not really an update, just a change from the default value. Better to say "In the Flow field, select **Application**."
- The instructions say to enter the token URL /oauth-endpoint-api/oauth2/token, but this doesn't match the screenshot.
- Re "Add a new Scope: view_current.", this is too terse, and there is nothing in the formatting to differentiate view_current as being the value that is entered for the name of the scope. Elsewhere in the tutorial, italics have been used for this purpose, so this needs to be consistent throughout the tutorial (and with the other tutorials)
- Having a step that just says "Save" is too terse.
- Re "Switch over to Drafts / Products", need to give the instructions for how to do this.
- Re "Add the OAuth Endpoint API to your Weather Provider product", need to give the instructions for how to do this.
- Re "stage it to your Sandbox", need to give instructions for how to do this.
- The screen capture indicates where the Save icon is. However, the user has already saved an API twice and no such indication was given , so this is inconsistent.

Step 3

- Lettered substeps rather than bullets.
- Occurrences of "select" should be "click"
- The first step doesn't read well, it says "publish your product" twice. There need to be clear instructions on how to publish a product. Also, the instructions are assuming that the navigation pane is pinned, but it might well not be, in which case the user wouldn't even see the Dashboard navigation item, they would have to open the navigation pane first, so the instructions need to cater for this.
- In the first two screen captures, there is no indication of what had to be clicked to make the menus display, and what has to be selected from those menus.
- there is wide variation in the resolutions of the screen captures across the tutorial, they all need to be the same.
- Your personal BM details in the screen capture need to be redacted (as in the earlier screen capture)

Step 4

- What is the significance of passing only the access token? Why is the user being asked to do this?
- Screen capture is huge, see comment above.
- Your personal BM details in the screen capture need to be redacted (as in the earlier screen capture)

Step 5

- The words "you previously did" are being displayed at the bottom of the image.
- Screen capture is huge, see comment above.
- Your personal BM details in the screen capture need to be redacted (as in the earlier screen capture)

As this is getting started, there needs to be a link(s) to the KC to direct the user to where they can get more detailed information. The following pages are ideal:
- https://www.ibm.com/support/knowledgecenter/SSMNED_5.0.0/com.ibm.apic.toolkit.doc/tutorial_apionprem_security_OAuth.html
- https://www.ibm.com/support/knowledgecenter/SSMNED_5.0.0/com.ibm.apic.toolkit.doc/capic_sec_api_oauth.html

**Comment by rgthelen** [comment]
Did most of the changes here.  A few of the suggestions were a bit ambiguous...
